src/bias/MetaD.cpp

```diff
@@ -877,7 +877,13 @@ MetaD::MetaD(const ActionOptions& ao):
 
   // MPI version
   parseFlag("WALKERS_MPI",walkers_mpi_);
-
+  
+  //If this Action is not compiled with MPI the user is informed and we exit gracefully
+  if(walkers_mpi_){
+    plumed_assert(Communicator::PlumedHasMPI) << "Invalid walkers configuration: WALKERS_MPI flag requires MPI compilation";
+    plumed_assert(Communicator::initialized()) << "Invalid walkers configuration: WALKERS_MPI needs the communicator correctly initialized.";
```

Comment:
Here I think you could just remove the first test.

Beside this, the way to raise the warnings is correct!!

Reply:
In my first draft i had a single error message with "MPI is not initialized or plumed is not compiled with MPI", but I think is more clear for the user to have two more specialized messages, and easier to test for the developers

Reply:
I think you are right